Examples/Swift/ViewController.swift

```diff
@@ -45,220 +60,203 @@ class ViewController: UIViewController, MGLMapViewDelegate, NavigationViewContro
-        lineCasing.lineJoin = MGLStyleValue(rawValue: NSValue(mglLineJoin: .round))
-        return lineCasing
+        let navigationViewController = NavigationViewController(for: route)
+        navigationViewController.simulatesLocationUpdates = simulationButton.isSelected
+        navigationViewController.navigationDelegate = self
+
+        present(navigationViewController, animated: true, completion: nil)
     }
     
-    /// Delegate method for changing the route line casing style
-    func navigationMapView(_ mapView: NavigationMapView, routeCasingStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer? {
-        let line = MGLLineStyleLayer(identifier: identifier, source: source)
+}
+
+extension ViewController: NavigationViewControllerDelegate {
+    func navigationViewController(_ navigationViewController: NavigationViewController, didArriveAt destination: MGLAnnotation) {
         
-        line.lineColor = MGLStyleValue(rawValue: UIColor(red:0.18, green:0.49, blue:0.78, alpha:1.0))
-        line.lineWidth = MGLStyleValue(rawValue: 8)
+        // Multiple waypoint demo
+        guard exampleMode == .multipleWaypoints, nextWaypoint != nil else { return }
+
+        // When the user arrives, present a view controller that prompts the user to continue to their next destination
+        // This typ of screen could show information about a destination, pickup/dropoff confirmation, instructions upon arrival, etc.
+        guard let confirmationController = self.storyboard?.instantiateViewController(withIdentifier: "waypointConfirmation") as? WaypointConfirmationViewController else { return }
+            
+        confirmationController.delegate = self
         
-        line.lineCap = MGLStyleValue(rawValue: NSValue(mglLineCap: .round))
-        line.lineJoin = MGLStyleValue(rawValue: NSValue(mglLineJoin: .round))
-        return line
+        navigationViewController.present(confirmationController, animated: true, completion: nil)
     }
-    
-    func navigationViewController(_ navigationViewController: NavigationViewController, didArriveAt destination: MGLAnnotation) {
-        print("User arrived at \(destination)")
+}
+
+extension ViewController: WaypointConfirmationViewControllerDelegate {
+    func confirmationControllerDidConfirm(controller confirmationController: WaypointConfirmationViewController) {
+        guard let nextDestination = nextWaypoint else { return }
+        guard let navigationViewController = self.presentedViewController as? NavigationViewController else { return }
+
+        // Calculate directions to the next waypoint
+        let options = RouteOptions(coordinates: [
+            navigationViewController.mapView!.userLocation!.coordinate,
+            nextDestination,
+        ])
+        options.includesSteps = true
+        options.routeShapeResolution = .full
+        options.profileIdentifier = .automobileAvoidingTraffic
```

Comment:
Can we get these details from the previous route’s `routeOptions`? That way, if the profile happens to be `.cycling`, it’ll automatically match when going to the next leg of the trip.